CMake FMUs with dependencies missing pthread DLL

### Description

The dynamic library `libwinpthread-1.dll` can't be found when creating FMUs with CMake and `--fmuRuntimeDepends=modelica`.

### Steps to Reproduce

Open the test `testsuite/omsimulator/cmakeFMU.mos` and change `--fmuRuntimeDepends=none` to `--fmuRuntimeDepends=modelica` and run the test

```bash
omc cmakeFMU.mos
```

Error:
```
CMake Error at cmake_install.cmake:71 (file):
  file Could not resolve runtime dependencies:

    libwinpthread-1.dll

make: *** [Makefile:100: install] Error 1
```

### Possible solution

Add something like
```modelica
Settings.getInstallationDirectoryPath() + "/bin"
```

to `locations` in [https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/Compiler/SimCode/SimCodeUtil.mo#L15569](SimCodeUtil.mo#L15569).

### Version and OS

 - OpenModelica Version: v1.20.0-dev-363
 - OS: Windows 10, 64 bit

### Additional Context

Related issue: https://github.com/OpenModelica/OpenModelica/issues/9509

--- a/OMCompiler/Compiler/SimCode/SimCodeUtil.mo
+++ b/OMCompiler/Compiler/SimCode/SimCodeUtil.mo
@@ -15564,6 +15564,7 @@ protected
   end addQuotationMarks;
 algorithm
   (locations, libraries) := getDirectoriesForDLLsFromLinkLibs(libs);
+  locations := listAppend({Settings.getInstallationDirectoryPath() + "/bin"}, locations);   // pthread located in OpenModelica/bin/ on Windows
   locations := List.map(locations, addQuotationMarks);
   // Use target_link_directories when CMake 3.13 is available and skip the find_library part
   cmakecode := cmakecode + "set(EXTERNAL_LIBDIRECTORIES " + stringDelimitList(locations, "\n                            ") + ")\n";

